examples/cpp/flash_attn/flash_attn.cu

```diff
@@ -0,0 +1,376 @@
+
+            compute::gemm_(rQ, rK, attn_block_f32);
+        }
+        load_rv(sV, rV);
+        __syncthreads();
+
+        ConvertAcc cast_acc;  // Convert acc to half precision
+        cast_acc(attn_block_f32, attn_block);
+
+        if (tiledcuda::thread(0)) {
+            printf("kRows: %d, kCols: %d\n", RegAccCast::kRows,
+                   RegAccCast::kCols);
+            printf("Thread 0 attn block: \n");
+            for (int h = 0; h < RegAccCast::kRows; ++h) {
+                for (int w = 0; w < RegAccCast::kCols; ++w) {
+                    printf("(%d, %d):\n", h, w);
+                    attn_block(h, w).dump_value();
+                }
+            }
+        }
+
+        // Copy `cur_max_vec`, `cur_norm_vec` into `prev_max_vec`,
+        // `prev_norm_vec`.
+        copy_vec(cur_max_vec, prev_max_vec);
+        copy_vec(cur_norm_vec, prev_norm_vec);
+
+        // Compute row max.
+        row_max(attn_block, cur_max_vec);
+
+        if (tiledcuda::thread(0)) {
+            // WarpLayout: (4, 1)
+            printf("Thread 0 cur_max_vec: \n");
+            cur_max_vec.dump_value();
+        }
+
+        // Broadcast subtract from `attn_block`.
+        broadcast_sub(cur_max_vec, attn_block);
+
+        if (tiledcuda::thread(0)) {
+            printf("Thread 0 attn block after broadcast sub: \n");
+            for (int h = 0; h < RegAccCast::kRows; ++h) {
+                for (int w = 0; w < RegAccCast::kCols; ++w) {
+                    printf("(%d, %d):\n", h, w);
+                    attn_block(h, w).dump_value();
+                }
+            }
+        }
+
+        // Compute exp in `attn_block`.
+        block_exp(attn_block, attn_block);
+
+        // Compute new max vector.
+        vec_max(cur_max_vec, prev_max_vec, new_max_vec);
+
+        // Renormalization for the previous block.
+        vec_sub(prev_max_vec, new_max_vec, prev_norm_vec);
+        vec_exp(prev_norm_vec, prev_norm_vec);
+
+        // Renormalization for the current block.
+        vec_sub(cur_max_vec, new_max_vec, cur_norm_vec);
+        vec_exp(cur_norm_vec, cur_norm_vec);
+
+        // Update normalization factor l(x)
+        vec_mul(prev_norm_vec, prev_sum_vec, prev_sum_vec);
+        vec_mul(cur_norm_vec, cur_sum_vec, cur_sum_vec);
+        vec_add(prev_sum_vec, cur_sum_vec, new_sum_vec);
+
+        // Compute unnormized attention block.
+        compute::gemm_(attn_block, rV, unnormized_attn_block_f32);
+
+        __syncthreads();
+
+        ConvertO cast_o;  // Convert half precision to float.
+        cast_o(unnormized_attn_block_f32, unnormized_attn_block);
+
+        vec_mul(prev_sum_vec, prev_norm_vec, prev_norm_vec);
+        broadcast_mul(prev_norm_vec, rO);
+
+        broadcast_mul(cur_norm_vec, unnormized_attn_block);
+
+        block_add(rO, unnormized_attn_block, rO);
+
+        // if (tiledcuda::thread0()) {
+        //     unnormized_attn_block.dump_value();
+        //     rO.dump_value();
+        // }
```

Comment:
At line 193, the debug print statements are commented out, while the debug print statements from lines 149 to 157 are still active. I suggest wrapping all debug print statements in a macro, as we have done before, to keep the code neat and maintain consistency.

```c
#ifdef DEBUG
...
#endif
```

Reply:
Fixed. And I added `ENABLE_DEBUG` option in CMake.